### ISSUE TITLE
Add option to opt out of modern CSS transform properties like `scale`, `translate`, etc.

### ISSUE THREAD

**Issue by brandonmcconnell** [sambauers/tailwindcss-3d#17]
Many of my users still use Safari v13.x, which does not support the newer CSS transform-related properties like `scale`, `translate`, etc.

By default, this is fine, as TailwindCSS supports the older transform styles, in this way:

```html
<div class="scale-150 rotate-45"></div>
```

```css
.rotate-45 {
    --tw-rotate: 45deg;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
}

.scale-150 {
    --tw-scale-x: 1.5;
    --tw-scale-y: 1.5;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
}
```

where each transform-related property applies the same foundational `transform` value:
```css
transform: 
  translate(var(--tw-translate-x), var(--tw-translate-y))
  rotate(var(--tw-rotate))
  skewX(var(--tw-skew-x))
  skewY(var(--tw-skew-y))
  scaleX(var(--tw-scale-x))
  scaleY(var(--tw-scale-y));
```

It'd be great if tailwindcss-3d could support 3d transforms but using the same syntax and impl-style as the existing TailwindCSS-native styles.

This could look something like this:

```html
<div class="scale-z-50 rotate-z-90 translate-z-44"></div>
```

```css
transform: 
  translate(var(--tw-translate-x), var(--tw-translate-y), var(--tw-translate-z))
  rotateX(var(--tw-rotate-x))
  rotateY(var(--tw-rotate-y))
  rotateZ(var(--tw-rotate, var(--tw-rotate)))
  skew(var(--tw-skew-x), var(--tw-skew-y))
  scale3d(var(--tw-scale-x, var(--tw-scale)), var(--tw-scale-y, var(--tw-scale)), var(--tw-scale-z, var(--tw-scale)));
```

**Comment by sambauers** [comment]
Hi Brandon,

Thanks for your thoughts and thorough request. Part of the intention of this plugin was to move away from the `transform` property as much as possible. The main benefit of doing this is that maintaining that set of `transform` options is extremely cumbersome, and in TailwindCSS core today, it is not possible (or at least very hard) to combine transforms as a result of events. This is especially true for animating elements. See this reduced example where a previously translated element fails to `animate-spin` on hover: https://play.tailwindcss.com/2wd29Wqtmx

While there are definitely drawbacks to the backwards compatibility you are proposing, it is probably fine as long as people are aware of the limitations. It will cause a bit of code overhead though, and I'm not sure how much. I will take a look to see if it can be done at minimal cost, but I can't commit yet to adding it to the plugin.

What I can do right away is provide a "Can I Use" panel on the readme so that people are aware of the limitation.

I'll update you here if and when this progresses.

**Comment by sambauers** [comment]
I've added a note on the README about browser compatibility: https://github.com/sambauers/tailwindcss-3d#browser-compatibility

**Comment by brandonmcconnell** [comment]
@sambauers Thanks! 🙌🏼

I think it might be a fairly low effort to incorporate the legacy version into this, optionally. I'm happy to investigate and open a PR if I figure that out.

**Comment by sambauers** [comment]
> I'm happy to investigate and open a PR if I figure that out.

I'm happy to collaborate on an approach to this, feel free to have a go and open a draft PR. My suggestion would be to solve it for just one of the utilities first, then once we land on a solution we can spread that to the other utilities.

**Comment by sambauers** [comment]
Released version 0.2.0 containing a first attempt at this. Please give it a run and report any issues.
